docs/troubleshooting.asciidoc

```diff
@@ -0,0 +1,236 @@
+
+[float]
+[id="{p}-pause-controllers"]
+=== Pause ECK controllers
+
+When debugging Elasticsearch, it may be necessary to "pause" the operator reconciliations so no resource gets modified or created in the meantime.
+
+It can be achieved by setting the annotation `common.k8s.elastic.co/pause` to `true` to any resource controlled by the operator :
+
+- Elasticsearch
+- Kibana
+- ApmServer
+
+[source,yaml]
+----
+metadata:
+  annotations:
+    common.k8s.elastic.co/pause: "true"
+----
+
+Or in one line:
+
+[source,sh]
+----
+kubectl annotate elasticsearch quickstart --overwrite common.k8s.elastic.co/pause=true
+----
+
+[float]
+[id="{p}-get-k8s-events"]
+=== Get Kubernetes events
+
+ECK will emit some events when:
+
+* important operations are performed (example: a new Elasticsearch Pod was created)
+* something is wrong, and the user should know about it
+
+Fetch Kubernetes events:
+
+[source,sh]
+----
+kubectl get events
+
+(...)
+28s       25m       58        elasticsearch-sample-es-p45nrjch29.15b3ae4cc4f7c00d   Pod                             Warning   FailedScheduling    default-scheduler                                         0/3 nodes are available: 1 node(s) had no available volume zone, 2 Insufficient memory.
+28s       25m       52        elasticsearch-sample-es-wxpnzfhqbt.15b3ae4d86bc269f   Pod                             Warning   FailedScheduling    default-scheduler                                         0/3 nodes are available: 1 node(s) had no available volume zone, 2 Insufficient memory.
+----
+
+The list of events can be filtered to a particular Elasticsearch cluster:
+
+[source,sh]
+----
+kubectl get event --namespace default --field-selector involvedObject.name=elasticsearch-sample
+
+LAST SEEN   FIRST SEEN   COUNT     NAME                                    KIND            SUBOBJECT   TYPE      REASON    SOURCE                     MESSAGE
+30m         30m          1         elasticsearch-sample.15b3ae303baa93c0   Elasticsearch               Normal    Created   elasticsearch-controller   Created pod elasticsearch-sample-es-4q7q2k8cl7
+30m         30m          1         elasticsearch-sample.15b3ae303bab4f40   Elasticsearch               Normal    Created   elasticsearch-controller   Created pod elasticsearch-sample-es-jg7dsfkcp8
+30m         30m          1         elasticsearch-sample.15b3ae303babdfc8   Elasticsearch               Normal    Created   elasticsearch-controller   Created pod elasticsearch-sample-es-xrxsp54jd5
+----
+
+This can also be done for Kibana and APM Server.
+
+[float]
+[id="{p}-exec-into-containers"]
+=== Exec into containers
+
+To troubleshoot a filesystem, configuration or network issue, it is sometimes useful to execute shell commands in the Elasticsearch container directly.
+This can be done with `kubectl`:
+
+[source,sh]
+----
+kubectl exec -ti elasticsearch-sample-es-p45nrjch29 bash
+----
+
+This can also be done for Kibana and APM Server.
+
+[float]
+[id="{p}-ask-for-help"]
+=== Ask for help
+
```

Comment:
Could be the place to mention the existence of `eck-dump.sh` but comes the question of the tar uploading?

Reply:
Yes, I wasn't sure about that one. I thought we could maybe wait a bit more to see what happens with community issues before we officially advertise it.